src/http-client.ts

```diff
@@ -12,23 +10,41 @@ export class HttpRequestError extends Error {
   }
 }
 
-export default class HttpClient {
-  constructor(private axiosInstance: AxiosInstance, private sdkParams: ISdkParams) {}
+export interface IHttpClient {
+  get<T>(resource: string): Promise<T | undefined>;
+}
+
+export default class FetchHttpClient implements IHttpClient {
+  constructor(private baseUrl: string, private sdkParams: ISdkParams, private timeout: number) {}
 
   async get<T>(resource: string): Promise<T | undefined> {
+    const url = new URL(this.baseUrl + resource);
+    Object.keys(this.sdkParams).forEach((key) =>
+      url.searchParams.append(key, this.sdkParams[key as keyof ISdkParams]),
+    );
+
     try {
-      const response = await this.axiosInstance.get<T>(resource, {
-        params: this.sdkParams,
-      });
-      return response.data;
+      // Canonical implementation of abortable fetch for interrupting when request takes longer than desired.
+      // https://developer.chrome.com/blog/abortable-fetch/#reacting_to_an_aborted_fetch
+      const controller = new AbortController();
+      const signal = controller.signal;
+      const timeoutId = setTimeout(() => controller.abort(), this.timeout);
+      const response = await fetch(url.toString(), { signal: signal });
+      // Clear timeout when response is received within the budget.
+      clearTimeout(timeoutId);
+
+      if (!response.ok) {
+        throw new HttpRequestError('Failed to fetch data', response.status);
+      }
+      return response.json() as Promise<T>;
     } catch (error) {
-      this.handleHttpError(error);
-    }
-  }
+      if (error.name === 'AbortError') {
+        throw new HttpRequestError('Request timed out', 408);
+      } else if (error instanceof HttpRequestError) {
+        throw error;
+      }
 
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  private handleHttpError(error: any) {
-    const status = error?.response?.status;
-    throw new HttpRequestError(error.message, status);
+      throw new HttpRequestError('Network error', 0);
```

Comment:
Kind of stinks this swallows the original error. How about modifying the signature to optionally take a third argument, cause, and then pass that along to Error.
```ts
super(message, { cause });
```